autogen

```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+cd "${0%/*}" || exit 1
+mkdir -p m4 build-aux || exit 1
```

Comment:
You could replace ``|| exit 1`` with ``set -e``.

Reply:
True. In general I prefer to explicitly exit (instead of `set -e`) for a few reasons:

  * If I'm in the habit of *always* doing `|| exit 1` then I'm more likely to notice when it's missing. If I change my habit to rely on `set -e` then I'm less likely to notice when `set -e` is missing in a script.
  * The POSIX specification for `set -e` is not very clear, and different shells do different things in corner cases, such as `eval false` and `x=$(false)`.
  * In more complicated scripts I like to add error messages:

    ```shell
    pecho() { printf %s\\n "$*"; }
    log() { pecho "$@"; }
    error() { log "ERROR: $@" >&2; }
    fatal() { error "$@"; exit 1; }

    foo bar || fatal "an error message goes here"
    ```

Reply:
I always set ``set -eou pipefail``.

Reply:
Unfortunately `-o pipefail` is not in [the POSIX standard](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#set) (it's a bashism).